### ISSUE TITLE
tf.nn.softmax fails with dim > 0

### ISSUE THREAD

**Issue by pekaalto** [tensorflow/tensorflow#6590]
If I do this:

    import tensorflow as tf
    import numpy as np

    session = tf.InteractiveSession()
    logits = np.array([[1.0,2.0,4.0],[1.0, 1.0, 1.0]])

    tf.nn.softmax(logits,dim=1).eval()

I get:
`InvalidArgumentError (see above for traceback): Requires start <= limit when delta > 0: 2/1
	 [[Node: range_9 = Range[Tidx=DT_INT32, _device="/job:localhost/replica:0/task:0/cpu:0"](range_9/start, Sub_6, range_9/delta)]]`

But both of these work:

    tf.nn.softmax(logits.T, dim=0).eval().T
    tf.nn.softmax(logits, dim=-1).eval()

Those are what I would expect to get from the first one with `dim=1`.

I am using '0.12.0-rc1'

**Comment by AllenAkhaumere** [comment]
Thick to what works.

**Comment by yaroslavvb** [comment]
Documentation is confusing. It says the default is the `last dimension`, which I would expect to be `dim=1` for a tensor with shape=(2,), @yuefengz can you comment?

**Comment by Response777** [comment]
came across with exactly the same problem. tf.__version__ = 1.0.1

**Comment by yuefengz** [comment]
This is already fixed.
